app/Models/Semester.php

```diff
@@ -152,18 +159,26 @@ public function getEndDate(): Carbon
         return Carbon::createFromDate($year, $month, 1)->endOfWeek();
     }
 
+    /**
+     * Whether the semester is in the past.
+     */
+    public function isClosed(): bool
```

Comment:
Hm... the problem is that only `SemesterEvaluationController` has the trait, not individual semesters. Or do we have some other means?

Reply:
You should use the SemesterEvaluationController's periodic event since it is connected to that.

Technically, you can also add the trait to this one and override the '$underlyingControllerName' so that this controller will use the same periodicEvent as the SemesterEvaluationController:

```
    use HasPeriodicEvent;

    function __construct()
    {
        $this->underlyingControllerName = SemesterEvaluationController::class;
    }
```
With this, I'd say rename the property to something like $periodicEventsAttachedController

Reply:
Can this return different end dates for different semesters? Or am I missing the point?

Reply:
Or can we also elect to consider all questions (including those of the previous semesters) as open when the evaluation form is open? Technically, it is not ideal, but it would simplify the code greatly while also not causing practical problems (they will probably only fill the questions on the form).

Reply:
With the suggested solution, the SemesterEvaluationController and the AnonymousQuestionsController will use the same PeriodicEvent, the. one that is connected and can be updated in SemesterEvaluationController.